### PR TITLE
New discriminated union example

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR adds a new discriminated union example.

### CODE REVIEW

**Review by pepone — APPROVED** [review]

**Review by ReeceHumphreys — APPROVED** [review]
Looks good!

**Review by InsertCreativityHere — APPROVED** [review]
Looks good!

File: examples/slice/DiscriminatedUnion/slice/Shape.slice [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    /// A rectangle.
+    /// @param width: The width of the rectangle.
+    /// @param height: The height of the rectangle.
+    Rectangle(width: float64, height: float64)
+
+    /// An ellipse.
+    /// @param majorAxis: The length of the major axis of the ellipse.
+    /// @param minorAxis: The length of the minor axis of the ellipse.
+    Ellipse(majorAxis: float64, minorAxis: float64)
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply examples/slice/DiscriminatedUnion/slice/Shape.slice]: Reworked the order, just not as suggested.

**Comment by bernardnormier** [comment]
For reference, the output is:

Client
```
The area of Rectangle { Width = 3, Height = 2 } is 6.00
The area of Ellipse { MajorAxis = 3, MinorAxis = 2 } is 4.71
The area of Square { Side = 4 } is 16.00
The area of Circle { Radius = 2 } is 12.57
```

Server
```
Computing area for shape Rectangle { Width = 3, Height = 2 }
Computing area for shape Ellipse { MajorAxis = 3, MinorAxis = 2 }
Computing area for shape Square { Side = 4 }
Computing area for shape Circle { Radius = 2 }
```

**Pull request merged**